jooby/src/main/java/io/jooby/problem/ProblemDetailsErrorHandler.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Jooby https://jooby.io
+ * Apache License Version 2.0 https://jooby.io/LICENSE.txt
+ * Copyright 2014 Edgar Espina
+ */
+package io.jooby.problem;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.jooby.*;
+import io.jooby.exception.*;
+import org.slf4j.Logger;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static io.jooby.MediaType.*;
+import static io.jooby.StatusCode.*;
+
+public class ProblemDetailsErrorHandler extends DefaultErrorHandler {
+
+  private boolean log4xxErrors;
+
+  public ProblemDetailsErrorHandler log4xxErrors() {
+    this.log4xxErrors = true;
+    return this;
+  }
+
+  @Override
+  public void apply(@NonNull Context ctx, @NonNull Throwable cause, @NonNull StatusCode code) {
+    Logger log = ctx.getRouter().getLog();
+    if (cause instanceof NotAcceptableException ex) {
```

Comment:
let's create an interface so exceptions can implement/convert to Problem

Reply:
If you need to create a custom problem you can do it by extending the `HttpProblem`. Here is an example https://github.com/kliushnichenko/jooby/blob/feat/problem-details/tests/src/test/java/io/jooby/problem/data/OutOfStockProblem.java

It is quite convenient to reuse the builder and it gives us enough control to guarantee that the problem instance is legit.
I'm afraid with an interface it won't be so convenient, and it for sure lacks control (for example the title is required and the status code should be valid). Also, static helper methods need to be moved somewhere...

It looks like there are more cons than pros. Or maybe I didn't get your point correctly

Reply:
I mean: 

```java

interface ToHttpProblem {
    HttpProblem toHttpProblem();
}
```

Probably a better name. So any exception can be converted to a HttpProblem (no need of custom message on handler) and when not we fallback to `HttpProblem.valueOf`